biogeochem/FatesAllometryMod.F90

```diff
@@ -1791,6 +1913,90 @@ subroutine dh2bagw_chave2014(d,h,dhdd,p1,p2,wood_density,c2b,bagw,dbagwdd)
+     !
+     ! -----------------
+     !  References
+     ! -----------------
+     ! Chave J, Rejou-Mechain M, Burquez A, Chidumayo E, Colgan MS, Delitti WB,
+     !    Duque A, Eid T, Fearnside PM, Goodman RC et al. 2014. Improved
+     !    allometric models to estimate the aboveground biomass of tropical
+     !    trees. Glob. Change Biol. 20: 3177-3190. doi:10.1111/gcb.12629.
+     !
+     ! Saldarriaga JG, West DC, Tharp ML , Uhl C. 1988. Long-term 
+     !    chronosequence of forest succession in the upper Rio Negro of
+     !    Colombia and Venezuela. J. Ecol. 76: 938-958.
+     !    doi:10.2307/2260625.
+     !
+     ! -----------------
+     !  Input arguments
+     ! -----------------
+     ! d            -- Diameter at breast height           [     cm]
+     ! h            -- Total tree height                   [      m]
+     ! dhdd         -- Height derivative with dbh          [   m/cm]
+     ! p1           -- Parameter 1 (log-intercept)         [     --]
+     ! p2           -- Parameter 2 (power, or log-slope)   [     --]
+     ! p3           -- Parameter 3 (power, or log-slope)   [     --]
+     ! wood_density -- Wood density                        [  g/cm3]
+     ! c2b          -- Carbon to biomass multiplier ~ 2    [ kg/kgC]
+     ! dbh_maxh     -- DBH at maximum height               [     cm]
+     !
+     ! ------------------
+     !  Output arguments
+     ! ------------------
+     ! bagw         -- Above-ground biomass per individual [    kgC]
+     ! dbagwdd      -- Above-ground biomass derivative     [ kgC/cm]
+     !
+     !--------------------------------------------------------------------------
+
+
+     !--- Arguments
+     real(r8), intent(in)            :: d            ! plant diameter              [    cm]
+     real(r8), intent(in)            :: h            ! plant height                [     m]
+     real(r8), intent(in)            :: dhdd         ! Height deriv. wrt diameter  [  m/cm]
+     real(r8), intent(in)            :: p1           ! Log-intercept parameter     [     -]
+     real(r8), intent(in)            :: p2           ! Log-slope parameter (size)  [     -]
+     real(r8), intent(in)            :: p3           ! Log-slope parameter (WD)    [     -]
+     real(r8), intent(in)            :: wood_density ! Wood density                [ g/cm3]
+     real(r8), intent(in)            :: c2b          ! Carbon to biomass factor    [kg/kgC]
+     real(r8), intent(out)           :: bagw         ! Above-ground biomass        [   kgC]
+     real(r8), intent(out), optional :: dbagwdd      ! AG biomass derivative       [kgC/cm]
+     !---~---
+
+
+     !--- Find above-ground biomass
+     bagw = p1 * (d*d*h)**p2 * wood_density**p3 / c2b
+     !---~---
+
+
+     !---~---
+     !   Compute the above-ground biomass derivative with DBH if needed, noting that
+     ! height is actually a function of DBH.
+     !---~---
+     if (present(dbagwdd)) then
+        dbagwdd = p2 * bagw * ( 2._r8 / d + dhdd / h )
```

Comment:
same as above, need to switch 2 with c2b?

Reply:
No, this is also due to the derivative.